src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -48,12 +42,33 @@ public async Task<string> GetSecretValue(Uri secretUri, CancellationToken cancel
 
             if (client != null)
             {
-                KeyVaultSecret secret = await client.GetSecretAsync(secretName, secretVersion, cancellationToken).ConfigureAwait(false);
-                secretValue = secret?.Value;
+                KeyVaultSecret secret;
+
+                // Try to load secret value from the cache first.
+                secretValue = GetCachedSecretValue(key);
+
+                if (secretValue == null)
+                {
+                    // We dont have a cached secret value or the cached value has expired.
+                    // Get the secret from Key Vault and update the cache.
+                    secret = await client.GetSecretAsync(secretName, secretVersion, cancellationToken).ConfigureAwait(false);
+                    secretValue = secret?.Value;
+
+                    if (secret != null)
+                    {
+                        UpdateCachedKeyVaultSecrets(key, secretValue);
```

Comment:
>UpdateCachedKeyVaultSecrets [](http://example.com/codeflow?start=24&length=27)

I would update that name to `Cache` to make that a bit easier to follow. From the name I thought there was a bit more going on at this step.

Reply:
Changed to  `SetSecretInCache`.